OPHD/Things/Structures/SeedPower.h

```diff
@@ -1,49 +1,36 @@
 #pragma once
 
 #include "Structure.h"
+#include "../../Constants.h"
 #include <string>
 
 const int SEED_POWER_PRODUCTION = 50;
 
-class SeedPower: public Structure
+class SeedPower: public PowerStructure
 {
 public:
-	SeedPower() :
-		Structure(constants::SEED_POWER, "structures/seed_1.sprite", StructureClass::CLASS_ENERGY_PRODUCTION)
+	SeedPower() : PowerStructure(constants::SEED_POWER, "structures/seed_1.sprite", StructureClass::CLASS_ENERGY_PRODUCTION)
 	{
 		sprite().play(constants::STRUCTURE_STATE_CONSTRUCTION);
 		maxAge(150);
 		turnsToBuild(5);
 		requiresCHAP(false);
 	}
 
-	StringTable createInspectorViewTable() override
-	{
-		StringTable stringTable(2, 1);
-
-		stringTable[{0, 0}].text = "Power Produced:";
-
-		auto energyProduced = calculateEnergyProduced();
-
-		stringTable[{1, 0}].text = std::to_string(energyProduced) + " / " + std::to_string(SEED_POWER_PRODUCTION);
-
-		if (energyProduced == 0)
-		{
-			stringTable[{1, 0}].textColor = constants::WARNING_TEXT_COLOR;
-		}
-
-		return stringTable;
-	}
-
 protected:
 	void defineResourceOutput() override
 	{
 		resourcesOut().energy(calculateEnergyProduced());
 	}
 
 private:
-	int calculateEnergyProduced()
+	int calculateEnergyProduced() override
 	{
 		return operational() ? SEED_POWER_PRODUCTION : 0;
 	}
+
+	int calculateMaxEnergyProduction() override
+	{
+		return SEED_POWER_PRODUCTION;
+	}
```

Comment:
You could implement `calculateEnergyProduced` in a generic way in the base class:
```cpp
	int calculateEnergyProduced() override
	{
		return operational() ? calculateMaxEnergyProduction() : 0;
	}
```

Reply:
I thought about this. I didn't know if power plants would eventually produce partial max power if they only had some of the needed resources available to consume or were damaged, etc. so left it as a pure virtual function.

Reply:
That could be neat.

Would the behaviour apply to all power plants, or only specific power plants?

Reply:
In particular, the Fusion Power Plant takes resources, so that was the one I was thinking about. 

Now that I'm thinking about it more, we can provide the default implementation as you suggested and override if needed in the child class. If that was your original suggestion, yet good idea, lets go with that. :)